src/sentry/workflow_engine/endpoints/validators.py

```diff
@@ -0,0 +1,148 @@
+from django.db import router, transaction
+from rest_framework import serializers
+from rest_framework.fields import Field
+
+from sentry import audit_log
+from sentry.api.serializers.rest_framework import CamelSnakeSerializer
+from sentry.issues import grouptype
+from sentry.issues.grouptype import GroupType
+from sentry.utils.audit import create_audit_entry
+from sentry.workflow_engine.models import DataConditionGroup, Detector
+from sentry.workflow_engine.models.data_condition import Condition, DataCondition
+from sentry.workflow_engine.types import DetectorPriorityLevel
+
+
+class BaseDataConditionValidator(CamelSnakeSerializer):
+
+    condition = serializers.CharField(
+        required=True,
+        max_length=200,
+        help_text="Condition used to compare data value to the stored comparison value",
+    )
+
+    @property
+    def comparison(self) -> Field:
+        raise NotImplementedError
+
+    @property
+    def condition_result(self) -> Field:
+        raise NotImplementedError
+
+    @property
+    def type(self) -> str:
+        # TODO: This should probably at least be an enum
+        raise NotImplementedError
+
+    def validate(self, attrs):
+        attrs = super().validate(attrs)
+        attrs["type"] = self.type
+        return attrs
+
+
+class NumericComparisonConditionValidator(BaseDataConditionValidator):
```

Comment:
This feels a bit more like a detector priority level condition, thoughts on renaming it to something like this?
```suggestion
class DetectorPriorityConditionValidator(BaseDataConditionValidator):
```

We could probably abstract a `NumericComparison` validator that's a bit more generic that this could use too 🤔

Reply:
This is a bit tricky because we'll probably want numeric comparators that map to other results, and detector priority results that want to work with other comparators. I don't have any great ideas on how to split this up.